coalib/misc/Caching.py

```diff
@@ -0,0 +1,81 @@
+import calendar
+import time
+import hashlib
+
+from coalib.output.Tagging import get_tags_dir as get_pickle_dir
+from coalib.misc.CachingUtilities import (
+    pickle_load, pickle_dump, time_consistent, update_time_db)
+
+
+class Cache:
```

Comment:
name it `FileCache` and remove every mention of `coala`, can you make this a python library seperate from coala? Or search if something like this is already available somewhere?